ci/ray_ci/pipeline/test_gap_filling_scheduler.py

```diff
@@ -0,0 +1,39 @@
+import sys
+from unittest import mock
+
+import pytest
+
+from ci.ray_ci.pipeline.gap_filling_scheduler import GapFillingScheduler
+
+
+@mock.patch(
+    "ci.ray_ci.pipeline.gap_filling_scheduler.GapFillingScheduler._get_latest_commits"
+)
```

Comment:
rather than mocking this, can we get a temp work dir, and create a real git repo for test there with fake history?

I prefer using real git repo for all git related operations and testing.

Reply:
do you want an e2e test? i think unit-test should mock the inputs; i can write a buildkite job as a e2e test once i have a binary to execute this e2e if you want